### ISSUE TITLE
[HOLD for payment 2025-01-15] [$250] Control Upgrade Pricing Not Displayed in Local Currency

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#53878]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:**  9.0.73-6
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @danielrvidal 
**Slack conversation** (hyperlinked to channel name): [expensify_convert](https://expensify.slack.com/archives/C07HPDRELLD/p1733803648585289)

## Action Performed:

1. Log in as a user, who is onboarding from the UK.
2. Navigate to the subscription page.
3. Observe that the subscription page pricing is displayed in GBP
4. Navigate Workspace and Upgrade to Control either by enabling rules or by connecting to Accounting
Full story link: https://fsty.io/v/hLcPr6Z8

## Expected Result:
All pricing, including the "Control" upgrade pricing, should be displayed in the user's local currency

## Actual Result:
The subscription page correctly displays pricing in GBP, but the "Control" upgrade pricing is incorrectly shown in USD.

## Workaround:
Unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details>

![image (11)](https://github.com/user-attachments/assets/cb3b0c19-8f11-45b9-9b01-eae60a14e6ea)

![Full Screen #43](https://github.com/user-attachments/assets/08d38ef5-a828-448e-a883-b744c559b360)

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021866573702240708931</li>
        <li>Upwork Job ID: 1866573702240708931</li>
        <li>Last Price Increase: 2024-12-10</li>
<li>Automatic offers: </li>
<ul>
<li>ahmedGaber93 | Reviewer | 105389816</li>
<li>Tony-MK | Contributor | 105389817</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @jliexpensify</details>

**Comment by Tony-MK** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Control Upgrade Pricing Not Displayed in Local Currency

### What is the root cause of that problem?

The problem is that we are hard-coding the prices of the subscription plans

https://github.com/Expensify/App/blob/499079b99b5b317bd6a8adae70e65faa35517acd/src/languages/en.ts#L4275-L4278

Then, we use it here.

https://github.com/Expensify/App/blob/499079b99b5b317bd6a8adae70e65faa35517acd/src/pages/workspace/upgrade/UpgradeIntro.tsx#L67-L73

### What changes do you think we should make in order to solve the problem?


Let's get the subscription price by referring to [`SUBSCRIPTION_PRICES`](https://github.com/Expensify/App/blob/499079b99b5b317bd6a8adae70e65faa35517acd/src/hooks/useSubscriptionPrice.ts#L7-L48) located in [`useSubscriptionPrice`](https://github.com/Expensify/App/blob/499079b99b5b317bd6a8adae70e65faa35517acd/src/hooks/useSubscriptionPrice.ts) to get the exact cost of the plan in the user's preferred currency.

https://github.com/Expensify/App/blob/499079b99b5b317bd6a8adae70e65faa35517acd/src/hooks/useSubscriptionPrice.ts#L7-L48

Therefore, we will replace this [line](https://github.com/Expensify/App/blob/499079b99b5b317bd6a8adae70e65faa35517acd/src/pages/workspace/upgrade/UpgradeIntro.tsx#L70) with something like this:

```
isCategorizing ? 
SUBSCRIPTION_PRICES[preferredCurrency][CONST.POLICY.TYPE.TEAM][CONST.SUBSCRIPTION.TYPE.PAYPERUSE]: 
SUBSCRIPTION_PRICES[preferredCurrency][CONST.POLICY.TYPE.CORPORATE][CONST.SUBSCRIPTION.TYPE.PAYPERUSE]
```

Also, if the user is from the UK, we should use GBP for the wallet balance instead of USD.

**Comment by jliexpensify** [comment]
This is a bug, Garrett mentioned we are [defaulting to USD](https://expensify.slack.com/archives/C07HPDRELLD/p1733841867114259?thread_ts=1733803648.585289&cid=C07HPDRELLD)

**Comment by ahmedGaber93** [comment]
Thanks all for the proposals.

@neonbhai `convertToShortDisplayString` is just a string method, not currency convertor, we should get the amount per currency first then pass it.

<br>

> {convertToShortDisplayString(
   SUBSCRIPTION_PRICES[preferredCurrency][upgradePlan][CONST.SUBSCRIPTION.TYPE.ANNUAL], preferredCurrency
)}

@Tony-MK this should work, but I think we can simplify it to be like below. WDYT?
```javascript

const preferredCurrency = usePreferredCurrency();
const subscriptionPrice = useSubscriptionPrice();

{convertToShortDisplayString(subscriptionPrice, preferredCurrency)}
```

**Comment by Tony-MK** [comment]
## Proposal
[Updated](https://github.com/Expensify/App/issues/53878#issuecomment-2532728828)

Hey @ahmedGaber93, `useSubscriptionPrice` only gets the price of the current subscription plan based on `useSubscriptionPlan`, and we want the price value of the upgraded subscription plan.

I updated my [proposal](https://github.com/Expensify/App/issues/53878#issuecomment-2532728828) account of this. WDYT?

**Comment by ahmedGaber93** [comment]
@Tony-MK Yeah, I got it. `useSubscriptionPrice` is not fit for that, let's back to your original proposal, it looks better.

**Comment by Tony-MK** [comment]
Done. Changed my proposal to look like the original proposal one.

**Comment by ahmedGaber93** [comment]
@Tony-MK's [proposal](https://github.com/Expensify/App/issues/53878#issuecomment-2532728828) using [SUBSCRIPTION_PRICES](https://github.com/Expensify/App/blob/499079b99b5b317bd6a8adae70e65faa35517acd/src/hooks/useSubscriptionPrice.ts#L7-L48) and `usePreferredCurrency()` LGTM!

🎀 👀 🎀 C+ reviewed

**Comment by jliexpensify** [comment]
Bumping @amyevans for a review! Also, I'm OOO until the 29th, but will keep this one assigned to myself. I don't believe payments will be needed before then!

**Comment by ahmedGaber93** [comment]
Not overdue, Waiting assigning

**Comment by amyevans** [comment]
Whoops, sorry I missed the initial assignment. @Tony-MK's proposal looks great.

**Comment by Tony-MK** [comment]
Hey @ahmedGaber93, the [PR](https://github.com/Expensify/App/pull/54340) is ready for review.

**Comment by jliexpensify** [comment]
Bumping @ahmedGaber93 for the checklist, thanks!

**Comment by jliexpensify** [comment]
Payment Summary

- @Tony-MK $250 (to be paid via ND)
- @ahmedGaber93 $250 (to be paid via ND) - can you also accept the invite? Cheers

https://www.upwork.com/jobs/~021866573702240708931

**Comment by ahmedGaber93** [comment]
> @ahmedGaber93 $250 (to be paid via ND) - can you also accept the invite? Cheers

@jliexpensify for me I will get paid via newDot. If there is an upwork invitation, please cancel it.

**Comment by jliexpensify** [comment]
Whoops sorry, fixed! Thanks for letting me know.

**Comment by jliexpensify** [comment]
Paid, and job closed.

@ahmedGaber93 - [here's the payment summary once again.](https://github.com/Expensify/App/issues/53878#issuecomment-2588188355)

**Comment by JmillsExpensify** [comment]
$250 approved for @ahmedGaber93
